src/components/text-input-with-copy-button/TextInputWithCopyButton.js

```diff
@@ -150,6 +150,9 @@ class TextInputWithCopyButton extends React.PureComponent<Props, State> {
     handleFocus = (event: SyntheticEvent<>) => {
         if (this.copyInputRef) {
             this.copyInputRef.select();
+            if (this.copyInputRef && this.copyInputRef.scrollLeft) {
```

Comment:
It looks like we access `this.copyInputRef` directly on the line above. Is the existence check necessary and, if so, should it be moved up?

Reply:
The reason for this check is for the flow error, for when the input is null. Adding this check stopped it from complaining

Reply:
It's already checked on line 151, so this code is unreachable if it's falsy. I'm not seeing a flow error when I make the change on my local. Can you try it once more?

Reply:
It looks like flow assumes that calling a method could destroy the object, even though the return of `.select()` isn't assigned to the class property. You can do something like this to get around it:

```
const { copyInputRef } = this;

if (copyInputRef) {
    copyInputRef.select();
    copyInputRef.scrollLeft = 0;
}
```

Hooray Flow.

Reply:
That's an interesting find. I wonder what would happen if this was using the createRef API instead of function refs

Reply:
That is interesting. I can definitely give that a try. 

Thats a good question Alex, we probably we should probably switch to the createRef api.